File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/interfaces/inheritance/view/InheritanceMainView.java
package com.nerdscorner.mvplib.testapp.interfaces.inheritance.view;

import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.nerdscorner.mvplib.interfaces.view.BaseActivityView;
import com.nerdscorner.mvplib.testapp.R;
import com.nerdscorner.mvplib.testapp.interfaces.inheritance.presenter.InheritanceMainPresenterInterface;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class InheritanceMainView extends BaseActivityView<InheritanceMainPresenterInterface> implements InheritanceMainViewInterface {
    @BindView(R.id.text) TextView textView;

    public InheritanceMainView(AppCompatActivity activity) {
        super(activity);
        ButterKnife.bind(this, activity);
    }

    @OnClick(R.id.some_button)
    public void onActionClicked() {
        getPresenter().onActionClicked();
    }

    @Override
    public void setTextValue(CharSequence value) {
        textView.setText(value);
    }
}

File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/inheritance/activities/view/InheritanceMainView.kt
package com.nerdscorner.mvplib.testapp.events.inheritance.activities.view

import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.nerdscorner.mvplib.events.view.BaseActivityView
import com.nerdscorner.mvplib.testapp.R

class InheritanceMainView(activity: AppCompatActivity) : BaseActivityView(activity) {
    var textView: TextView = activity.findViewById(R.id.text)

    init {
        onClick(R.id.some_button, R.id.some_button, event = ActionClickedEvent())
    }

    fun setTextValue(value: CharSequence) {
        textView.text = value
    }

    class ActionClickedEvent
}
